test/smoke/dashCore.js

```diff
@@ -1,18 +1,115 @@
-describe('DashCore', () => {
-  describe('All nodes', () => {
-    // `subversion` from `getNetworkInfo` response.
-    it('should have correct network type');
-
-    // The number of blocks should be almost the same (-3/+3) and block hash of particular
-    // block height should be the same. Using `blocks` and `bestblockhash` from `GetBlockChainInfo`.
-    it('should propagate blocks');
+const RpcClient = require('@dashevo/dashd-rpc/promise');
+
+const getNetworkConfig = require('../../lib/test/getNetworkConfig');
+
+const networkConfig = getNetworkConfig();
+
+
+describe('All nodes', () => {
+  let nodeNames = networkConfig.inventory.masternodes.hosts;
+  nodeNames = nodeNames.concat(networkConfig.inventory['wallet-nodes'].hosts);
+  // TODO miner rpc call stuck. why?
+  // nodeNames = nodeNames.concat(networkConfig.inventory['miners'].hosts);
+  nodeNames = nodeNames.concat(networkConfig.inventory['full-nodes'].hosts);
+  nodeNames.forEach((nodeName) => {
+    describe(nodeName, () => {
+      const config = {
+        protocol: 'http',
+        user: 'dashrpc',
+        pass: 'password',
+        host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+        port: 20002,
+      };
+      const rpc = new RpcClient(config);
+
+      it('should have correct network type', async () => {
+        const networkInfo = await rpc.getNetworkInfo();
+        expect(networkInfo.error).to.be.equal(null);
```

Comment:
RpcClient should throw an error if got `error` so that check doesn't make sense.

Reply:
removed